### PR TITLE
lightning: trim prefix when use gcs/s3 Storage WalkDir

### LINKED ISSUE #34118: lightning: extra slash before gcs files
## Bug Report
import using gcs storage, address like: `gcs://xxx_test/xxx_test3?region=ap-northeast-1`, with no trailing slash, on import:
```
file is filtered by file router"] [path=/test.t1.000000000.csv]
```
configing pattern to: `pattern: ^test\\.t.*$`. has to add another `.*` to the beginning of the pattern to make it import.

### 2. What did you expect to see? (Required)
match the file
### 3. What did you see instead (Required)
failed to match
### 4. What is your TiDB version? (Required)
lightning 5.3.1

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

Issue Number: close #34118 

Problem Summary:

### What is changed and how it works?

Trim right Prefix 

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by D3Hunter — APPROVED** [review]
lgtm

File: br/pkg/storage/gcs.go [hunk 1]
```diff
@@ -203,7 +201,7 @@ func (s *gcsStorage) WalkDir(ctx context.Context, opt *WalkOption, fn func(strin
 		// when walk on specify directory, the result include storage.Prefix,
 		// which can not be reuse in other API(Open/Read) directly.
 		// so we use TrimPrefix to filter Prefix for next Open/Read.
-		path := strings.TrimPrefix(attrs.Name, s.gcs.Prefix)
+		path := strings.TrimPrefix(attrs.Name, prefix)
```
Comment (gozssky) [inline br/pkg/storage/gcs.go:204]: Could you add a test for it?

**Comment by Ehco1996** [comment]
ptal @gozssky @3pointer 🧡

**Review by gozssky — APPROVED** [review]

File: br/pkg/storage/gcs.go (hunk 1 above)
Comment (3pointer) [reply br/pkg/storage/gcs.go:204]: I'm afraid `prefix` isn't equal to `s.gcs.Prefix` when `opt.SubDir` is set

Comment (Ehco1996) [reply br/pkg/storage/gcs.go]: thanks you are right, I should not change the default behavior here, now this PR just adds more test cases for `WalkDir` ptal @3pointer

Comment (gozssky) [reply br/pkg/storage/gcs.go]: Can we trim the extra slash prefix for the path?  It seems the behaviors of `gcs` and `s3` are not consistent with `local`.

Comment (Ehco1996) [reply br/pkg/storage/gcs.go]: ping @gozssky  do you have more suggestions? if not I think this pr is ready to merge?

Comment (gozssky) [reply br/pkg/storage/gcs.go]: LGTM. Please @3pointer take a look again.

**Review by 3pointer — APPROVED** [review]

**Pull request merged**